constd/main.go

```diff
@@ -46,70 +38,71 @@ func main() {
 
 	c := &constd{
 		logger: l,
-		config: config,
+		config: conf,
 		atmo:   newWatcher("atmo", l),
 		sats:   map[string]*watcher{},
 	}
 
 	var appSource appsource.AppSource
-	var errchan chan error
+	var errChan chan error
 
 	// if an external control plane hasn't been set, act as the control plane
 	// but if one has been set, use it (and launch all children with it configured)
-	if c.config.controlPlane == defaultControlPlane {
-		appSource, errchan = startAppSourceServer(config.bundlePath)
+	if c.config.ControlPlane == config.DefaultControlPlane {
+		appSource, errChan = startAppSourceServer(c.config.BundlePath)
 	} else {
-		appSource = appsource.NewHTTPSource(c.config.controlPlane)
+		appSource = appsource.NewHTTPSource(c.config.ControlPlane)
 
 		if err := startAppSourceWithRetry(l, appSource); err != nil {
 			log.Fatal(errors.Wrap(err, "failed to startAppSourceHTTPClient"))
 		}
 
-		if err := registerWithControlPlane(c.config); err != nil {
+		if err = registerWithControlPlane(c.config); err != nil {
 			log.Fatal(errors.Wrap(err, "failed to registerWithControlPlane"))
 		}
 
-		errchan = make(chan error)
+		errChan = make(chan error)
```

Comment:
I'd suggest pulling the creation of the appsource and errchan out into a function like `func setupControlPlane(config *config.Config) (appsource.AppSource, errChan) {...}`

Reply:
Broke it out in https://github.com/suborbital/sat/pull/62/commits/5ea5140a30317956c26c54e7134e6e2ae3ade9c6